operators/src/pro/cache/shared_cache.rs

```diff
@@ -0,0 +1,1190 @@
+        self.vector_caches
+            .get_mut(key)
+            .map(|op| OperatorCacheEntryView {
+                operator_cache: op,
+                cache_size: &mut self.cache_size,
+                landing_zone_size: &mut self.landing_zone_size,
+                lru: &mut self.lru,
+            })
+    }
+}
+
+impl CacheView<RasterCacheQueryEntry, RasterLandingQueryEntry> for CacheBackend {
+    fn operator_caches_mut(
+        &mut self,
+    ) -> &mut HashMap<CanonicOperatorName, RasterOperatorCacheEntry> {
+        &mut self.raster_caches
+    }
+}
+
+impl CacheView<VectorCacheQueryEntry, VectorLandingQueryEntry> for CacheBackend {
+    fn operator_caches_mut(
+        &mut self,
+    ) -> &mut HashMap<CanonicOperatorName, VectorOperatorCacheEntry> {
+        &mut self.vector_caches
+    }
+}
+
+pub trait CacheElement: ByteSize + Send + ByteSize
+where
+    Self: Sized,
+{
+    type Query: CacheQueryMatch + Clone + Send + Sync;
+    type LandingZoneContainer: LandingZoneElementsContainer<Self>;
+    type CacheContainer: CacheElementsContainer<Self::Query, Self, ResultStream = Self::ResultStream>
+        + From<Self::LandingZoneContainer>;
+    type ResultStream;
+    type CacheElementSubType: CacheElementSubType<CacheElementType = Self>;
+
+    fn move_into_landing_zone(
+        self,
+        landing_zone: &mut Self::LandingZoneContainer,
+    ) -> Result<(), CacheError> {
+        landing_zone.insert_element(self)
+    }
+
+    fn update_stored_query(&self, query: &mut Self::Query) -> Result<(), CacheError>;
+
+    fn cache_hint(&self) -> CacheHint;
+
+    fn typed_canonical_operator_name(key: CanonicOperatorName) -> TypedCanonicOperatorName;
+}
+
+pub trait CacheElementSubType {
+    type CacheElementType: CacheElement;
+
+    fn insert_element_into_landing_zone(
+        landing_zone: &mut <Self::CacheElementType as CacheElement>::LandingZoneContainer,
+        element: Self::CacheElementType,
+    ) -> Result<(), super::error::CacheError>;
+
+    fn create_empty_landing_zone() -> <Self::CacheElementType as CacheElement>::LandingZoneContainer;
+
+    fn result_stream(
+        cache_elements_container: &<Self::CacheElementType as CacheElement>::CacheContainer,
+        query: &<Self::CacheElementType as CacheElement>::Query,
+    ) -> Option<<Self::CacheElementType as CacheElement>::ResultStream>;
+}
+
+#[derive(Debug)]
+pub struct SharedCache {
+    backend: RwLock<CacheBackend>,
+}
+
+impl SharedCache {
+    pub fn new(cache_size_in_mb: usize, landing_zone_ratio: f64) -> Result<Self> {
+        if landing_zone_ratio <= 0.0 {
+            return Err(crate::error::Error::QueryingProcessorFailed {
+                source: Box::new(CacheError::LandingZoneRatioMustBeLargerThanZero),
+            });
+        }
+
+        if landing_zone_ratio >= 0.5 {
+            return Err(crate::error::Error::QueryingProcessorFailed {
+                source: Box::new(CacheError::LandingZoneRatioMustBeSmallerThenHalfCacheSize),
```

Comment:
couldn't it be also 50750?

Reply:
to be honest. I think it can't if we fill the landing zone  up to 50% of the cache size then it will not fit into the cache since there are some bytes used by the Arcs